### ISSUE TITLE
Can't pass null to find_circles_grid

### ISSUE THREAD

**Issue by 0e4ef622** [twistedfall/opencv-rust#605]
OpenCV lets you pass `null` for the `blobDetector` parameter of [`findCirclesGrid`](https://docs.opencv.org/4.x/d9/d0c/group__calib3d.html#ga7f02cd21c8352142890190227628fa80) but this does not appear possible in rust, at least not without unsafe.

**Comment by twistedfall** [comment]
Thanks for reporting this, I'll fix in in the upcoming release!

**Comment by Abrahamh08** [comment]
> Thanks for reporting this, I'll fix in in the upcoming release!

```
let nullf2d;
unsafe {
    nullf2d = Feature2D::from_raw(core::ptr::null_mut());
}
find_circles_grid_1(&centers, board_size, &mut out_centers, CALIB_CB_SYMMETRIC_GRID, &Ptr::new(nullf2d)).unwrap();
```

Passing null as the Feature2D instead of Ptr::from_raw works by the way. However, it would be appreciated if you could override the arg to be nullable. I tried it, but then there's some weirdness with &Ptr where it becomes Option<&Option<Ptr>> instead of Option<Ptr> or whatever it should be in this case

**Comment by twistedfall** [comment]
This is now fixed in 0.93.0

**Comment by Abrahamh08** [comment]
> This is now fixed in 0.93.0

I can now pass None, but I get a STATUS_ACCESS_VIOLATION

**Comment by twistedfall** [comment]
@Abrahamh08 Can you please provide a minimum example I can use to reproduce the error you're seeing?

**Comment by 0e4ef622** [comment]
```rs
use opencv::{
    calib3d::{find_circles_grid_1, CALIB_CB_SYMMETRIC_GRID},
    core::{Point2f, Size, Vector},
};

fn main() {
    let mut centers = Vector::<Point2f>::new();
    for i in 0..4 {
        for j in 0..4 {
            centers.push((i as f32 * 100.0, j as f32 * 100.0).into());
        }
    }

    let mut out_centers = Vector::<Point2f>::new();
    find_circles_grid_1(
        &centers,
        Size::new(4, 4),
        &mut out_centers,
        CALIB_CB_SYMMETRIC_GRID,
        None,
    )
    .unwrap();
}
```

**Comment by 0e4ef622** [comment]
Also for reference, this works

```rs
use opencv::{
    calib3d::{find_circles_grid_1, CALIB_CB_SYMMETRIC_GRID},
    core::{Point2f, Ptr, Size, Vector}, features2d::Feature2D, traits::Boxed,
};

fn main() {
    let mut centers = Vector::<Point2f>::new();
    for i in 0..4 {
        for j in 0..4 {
            centers.push((i as f32 * 100.0, j as f32 * 100.0).into());
        }
    }

    let mut out_centers = Vector::<Point2f>::new();
    let nullf2d;
    unsafe {
        nullf2d = Feature2D::from_raw(core::ptr::null_mut());
    }
    let r = find_circles_grid_1(
        &centers,
        Size::new(4, 4),
        &mut out_centers,
        CALIB_CB_SYMMETRIC_GRID,
        Some(&Ptr::new(nullf2d)),
    )
    .unwrap();
    dbg!(r);
    dbg!(out_centers);
}
```

**Comment by twistedfall** [comment]
Thank you, I can see why it happens and will try to provide a proper fix soon

**Comment by twistedfall** [comment]
Should now definitely be fixed in 0.93.3

**Comment by 0e4ef622** [comment]
Can confirm it is fixed, thanks!
